renderlib/FileReaderTIFF.cpp

```diff
@@ -317,21 +321,56 @@ readTiffDimensions(TIFF* tiff, const std::string filepath, VolumeDimensions& dim
 
 // return number of bytes copied to dest
 size_t
-copyAndConvert(uint8_t* dest, const uint8_t* src, size_t numBytes, int srcBitsPerPixel)
+copyDirect(uint8_t* dest, const uint8_t* src, size_t numBytes, int srcBitsPerPixel)
+{
+  memcpy(dest, src, numBytes);
+  return numBytes;
+}
+
+// convert pixels
+// this assumes tight packing of pixels in both buf(source) and dataptr(dest)
+// assumes dest is of format IN_MEMORY_BPP
+// return 1 for successful conversion, 0 on failure (e.g. unacceptable srcBitsPerPixel)
+size_t
+convertChannelData(uint8_t* dest, const uint8_t* src, const VolumeDimensions& dims)
 {
+  // how many pixels in this channel:
+  size_t numPixels = dims.sizeX * dims.sizeY * dims.sizeZ;
+  int srcBitsPerPixel = dims.bitsPerPixel;
+
   // dest bits per pixel is IN_MEMORY_BPP which is currently 16, or 2 bytes
   if (IN_MEMORY_BPP == srcBitsPerPixel) {
-    memcpy(dest, src, numBytes);
-    return numBytes;
+    memcpy(dest, src, numPixels * (srcBitsPerPixel / 8));
+    return 1;
   } else if (srcBitsPerPixel == 8) {
-    // convert pixels
-    // this assumes tight packing of pixels in both buf(source) and dataptr(dest)
     uint16_t* dataptr16 = reinterpret_cast<uint16_t*>(dest);
-    for (size_t b = 0; b < numBytes; ++b) {
+    for (size_t b = 0; b < numPixels; ++b) {
```

Comment:
I'm assuming you're using a uint16 to copy 2 at a time? 
What if you have an odd number of 8-bit pixels? 
Why are you storing an 8-bit pixel in a uint16?

Reply:
This function is converting everything to 16-bit uints.  This code branch is converting 8-bit uints (src) to 16-bit uints (dest).